synfig-studio/src/gui/widgets/widget_soundwave.cpp

```diff
@@ -333,11 +335,18 @@ bool Widget_SoundWave::do_load(const synfig::filesystem::Path& filename)
 		}
 		if (buffer.empty()) {
 			synfig::warning("sound frame #%i got empty buffer", i);
-			int buffer_length = (end_frame - start_frame + 1) * _channels * bytes_per_sample * std::round(_frequency/fps);
+			int buffer_length = (end_frame - start_frame) * _channels * bytes_per_sample * std::round(_frequency/fps);
 			buffer.resize(buffer_length);
 		}
 		int _n_bytes = _n_samples * _channels * bytes_per_sample;
-		std::copy(static_cast<unsigned char*>(_buffer), static_cast<unsigned char*>(_buffer) + _n_bytes, buffer.begin()+bytes_written);
+		if (bytes_written + _n_bytes > buffer.size()) {
+			if (buffer.size() <= bytes_written) {
+				synfig::error(_("Internal error: Widget_SoundWave: trying to read more bytes than buffer size: %i x %zu"), _n_bytes + bytes_written, buffer.size());
+				break;
```

Comment:
`delete frame;`?

Reply:
Yes!
But then I use unique_ptr instead just in case.
Or do you think it would be better the delete (I don't know how it affects performance)

Reply:
For unique_ptr, there is no difference, so it looks better in my opinion